text/0321-key-functions.md

```diff
@@ -0,0 +1,74 @@
+- Start Date: 2018-03-24
+- RFC PR: https://github.com/emberjs/rfcs/pull/321
+- Ember Issue: (leave this empty)
+
+# Key Functions for `each`/`each-in`
+
+## Summary
+
+This RFC proposes that we support passing functions to the `key` argument of the `each` and `each-in` helpers. These functions would be called for each item and would be expected to return they key value used in diffing.
+
+For example,
+
+```js
+import Component from '@ember/component';
+
+export default Component.extend({
+  itemKey(item) {
+    return `${item.factoryNumber}-${item.partNumber}`;
+  }
+});
+```
+
+```hbs
+{{#each @items key=itemKey as |item|}}
+  {{item-details item=item}}
+{{/each}}
+```
+
+## Motivation
+
+The `key` argument to `each` and `each-in` determines the key used in the Glimmer diffing algorithm. Here is a description of the currently accepted values?
```

Comment:
The ending question mark seems pretty out of place here. Is this a description of the current support, or is the RFC asking for clarification?

Reply:
https://www.youtube.com/watch?v=tqNhEzrWQpY

Reply:
Yeah, I don't know what I was thinking?